### PR TITLE
feat: add OS-level tool isolation with Seatbelt and Bubblewrap + Refactor config loaders

### PR DESCRIPTION
**Pull request by midodimori**
Security features:
- Seatbelt (macOS) and Bubblewrap (Linux) sandbox backends
- Deny-by-default permission model with FILESYSTEM and NETWORK permissions
- Permission intersection: tools get only what they need, capped by sandbox policy
- Thread-safe profile management with cleanup

Implementation:
- SandboxMiddleware intercepts tool calls and enforces permissions
- Worker process executes tools in isolated subprocess
- shlex-based offline flag injection for npx/uvx commands
- JSON protocol for parent-worker communication
- Package caching support for offline sandbox execution

Configuration:
- Sandbox profiles in .langrepl/sandboxes/*.yml
- Per-tool permissions via metadata["sandbox_permissions"]
- Per-MCP server permissions in config.mcp.json

Testing:
- 15 new test files covering permissions, execution, and edge cases
- Integration tests for end-to-end sandbox execution

## Summary by CodeRabbit

* **New Features**
  * OS-level sandbox execution (macOS Seatbelt, Linux Bubblewrap) with per-agent/tool selection, configurable permissions, and timeouts.

* **UX**
  * Sandbox marker shown for sandboxed tool calls; new CLI verbose flag for runtime logs; per-tool sandbox-bypass for trusted tools.

* **Documentation**
  * Expanded docs and examples (YAML) covering sandbox configs, permission semantics, paths, and behavior notes.

* **Tests / CI**
  * New sandbox unit & integration tests, CI workflow for integration, and make targets for unit vs integration.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/configs/checkpointer.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    @property
+    def checkpointer_names(self) -> list[str]:
+        return [cp.type for cp in self.checkpointers]
+
+    def get_checkpointer_config(
+        self, checkpointer_name: str
+    ) -> CheckpointerConfig | None:
+        return next(
+            (cp for cp in self.checkpointers if cp.type == checkpointer_name), None
+        )
```
[the remark this answers is not in the document]
Comment (midodimori) [reply src/configs/checkpointer.py]: class CheckpointerProvider(str, Enum)

File: README.md [hunk 2]
```diff
@@ -501,13 +513,28 @@ agents:
...
 
+   **Metadata fields**:
+   | Field | Description |
+   |-------|-------------|
+   | `approval_config.name_only` | Only match tool name for approval (ignore args) |
+   | `approval_config.format_args_fn` | Function to transform args for approval matching |
+   | `approval_config.render_args_fn` | Function to render args for display |
+   | `sandbox_permissions` | Required permissions: `FILESYSTEM`, `NETWORK`, or both |
+   | `sandbox_bypass` | Set `True` to skip sandbox entirely |
+
```
[the remark this answers is not in the document]
Comment (midodimori) [reply README.md]: no
